README.md

```diff
@@ -41,3 +41,26 @@ This project is heavily inspired by [johalun/sysctl-rs](https://github.com/johal
 Kudos to @fabianfreyer for untiring support and debugging.
 
 It is developed and maintained by Stefan Grönke ([@gronke](https://github.com/gronke)) and published under [BSD 2-Clause License](LICENSE.txt).
+
+## Development
+
+### Unit Tests
+
+Unit tests may run on FreeBSD or HardenedBSD.
+
+### Static Code Analysis
+
+The project enforces PEP-8 code style and MyPy strong typing via flake8, that is required to pass before merging any changes.
+Together with Bandit checks for common security issues the static code analysis can be ran on Linux and BSD code execution.
+
+```
+make install-dev
+make check
+```
+
+### Releases
+
+We try to *manually* keep a [Changelog](CHANGELOG.md), following the style on [changelog.md](https://changelog.md).
+New releases are tagged according to [Semver](https://semver.org/), released on [PyPi](https://pypi.org/project/libioc/), and published as [port](https://github.com/bsdci/ports).
+
+To get a port published, we need to [create a Bugzilla Issue in the Ports category](https://bugs.freebsd.org/bugzilla/enter_bug.cgi?component=Individual%20Port%28s%29&product=Ports%20%26%20Packages)
```

Comment:
I think we should automate the release and changelog process. Instead of tracking changes manually, we could use the GitHub Pull-Request titles as bullets in the changelog. A git release tag message should contain the latest changelog release section.

You could transfer the entire **Releases** section to an GitHub Issue instead of putting it in the readme file.

Reply:
> I think we should automate the release and changelog process. Instead of tracking changes manually, we could use the GitHub Pull-Request titles as bullets in the changelog.

that tends to produce a horrible Changelog that's hard to read…
i would understand this if we had 30 projects to release, but with 3.

Reply:
> that tends to produce a horrible Changelog that's hard to read…

So maybe we don't want to automatically write the entire messages, but automate creating a release PR (and keeping it up to date after rebasing) with the messages suggested as Issue Body. When clicking merge there, this message should be put in CHANGELOG.md automatically - together with a version bump.